src/shared/data-value-mutations/use-api-error.js

```diff
@@ -1,27 +1,75 @@
 import { useAlert } from '@dhis2/app-runtime'
-
 /**
- * A dictionary taking the shape Dictionary<ErrorCode, String> to override the display message for a certain error code
+ * A dictionary taking the shape A dictionary taking the shape Dictionary<ErrorCode, AlertProps> to override the display message for a certain error code
  * Potential error codes can be found here: https://github.com/dhis2/dhis2-core/blob/28a6aa052f221626e583a84bc18d80c0a8fa0927/dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ErrorCode.java#L97
  */
-const customErrorMessages = {
-    // E2017: i18n.t('Here we could override the message for E2017 error with a custom message in the FE')
+const customErrorAlertProps = {
+    // E2017: { message: i18n.t('Here we could override the message for E2017 error with a custom message in the FE') }
 }
 
-const useApiError = () => {
+/**
+ * A dictionary taking the shape Dictionary<ErrorCode, AlertProps>
+ */
+const statusCodeErrorAlertProps = {
+    // this is handled in `retry`
+    // 401: {
+    //     message: i18n.t(
+    //         'Session Expired. Please reload to log in again. You will not lose any data.'
+    //     ),
+    //     actions: [{ label: 'Reload now', onClick: () => location.reload() }],
+    // },
+}
```

Comment:
I refactored the structure of this before I realised the session-expiration handling had to be done in `retry` instead of `onError`. 

Should I delete this refactor? It does make the overrides more flexible for the future, but also introduces a bit more complexity.

Reply:
Note that this part will probably be changed a bit in a new PR handling "permanent" submit errors.